### DISCUSSION TITLE
Difference between Holochain Dev kit, HoloPort, Holo Launcher

### DISCUSSION THREAD

**Discussion by wadro** [holochain/holochain#1422] [Q&A]
Hi, I'm looking holochain project interesting. 
But I'm confused of three concept.
There are two things, [Holochain Dev Kit](https://developer.holochain.org/install/), [Holo Launcher](https://github.com/holochain/launcher), which I can install from browser.
And also there is Holo host, the hardsware which called [HoloPort](https://holo.host/host/).

So, the questions,
1. I think holochain source code in this repository is the same thing with the code in the HoloPort harware. Is it right?
2. Why does minimum requirement need for Holochain Dev Kit? I think it is like Android Studio or Xcode, JDK... 
3. What's the difference between holochain source code and HDK code? 
Are these two things combine and form a structure like this? Layer 1: Holochain, Layer 2: hApp (like application layer in tcp/ip)
4. What is Holo Launcher? It says hApp installer, builder(package) and executer, so it has DHT and holochain function in itself? Could I consider it as a execute file of holochain source code which is same with the code on the HoloPort in my question1? or is same with the code on HDK in my question2? 

    4-1. I read Holo Launcher is _Conductor_. I read this [document](https://holochain.github.io/holochain-rust/conductors.html) briefly, 
Is its funciton(conductor) inside holochain source code or inside HDK code? or is it a combination of hApps?

5. If question4 is correct, Can I run a holochain node with launcher(or HDK) without HoloPort? How can I contribute or participate in holochain network as running a node? 

If all of my questions solved, I want to dig into holochain more further.
Thanks for reading my dumb questions.

**Comment by f13end** [comment]
@guillemcordoba
